src/components/MainContent/MainContent.jsx

```diff
@@ -1,11 +1,13 @@
 import { Route, Routes } from "react-router-dom"
 import ArticlesSection from "../ArticlesSection/ArticlesSection"
+import SingleArticle from "../SingleArticle/SingleArticle"
 
 const MainContent = () => {
   return (
     <Routes>
       <Route path="/" element={<ArticlesSection />} />
       <Route path="/articles" element={<ArticlesSection />} />
+      <Route path="/:article" element={<SingleArticle />} />
     </Routes>
```

Comment:
The URL should be /articles/:article_id.

If you were to add another route below path="/:article" you would never reach it as what you have is a parametric endpoint.

Reply:
Thanks for pointing it out! In this case, this is how I would refactor it, if you agree that the title part of URL can stay.
```suggestion
        <Route path="/articles/:topic/:article_id/:article_title" element={<SingleArticle />} />
    </Routes>
```